### PR TITLE
Touch events

### PR DESCRIPTION
**Pull request by irenesmith**
A checklist to help your pull request get merged faster:
- Added BCD for Touch events to Document
- Added BCD for Touch events to Element

This is part refactoring touch events - see [ZenHub #950](https://app.zenhub.com/workspaces/mdn-sprint-board-5ab3d23fd0f4ea53b0022a61/issues/mdn/sprints/950)

### CODE REVIEW

File: api/Document.json [hunk 1]
```diff
@@ -7327,6 +7327,214 @@
...
+              "version_added": "22"
+            },
+            "chrome_android": {
+              "version_added": "25"
+            },
+            "edge": {
+              "version_added": "12"
+            },
+            "edge_mobile": {
+              "version_added": "6"
```
Comment (vinyldarkscratch) [inline api/Document.json]:
```suggestion
              "version_added": "12”
```

File: api/Document.json [hunk 2]
```diff
@@ -7327,6 +7327,214 @@
...
```
Comment (vinyldarkscratch) [inline api/Document.json]:
```suggestion
              "version_added": "12"
```

File: api/Document.json [hunk 3]
```diff
@@ -7327,6 +7327,214 @@
...
```
Comment (vinyldarkscratch) [inline api/Document.json]:
```suggestion
              "version_added": "12"
```

File: api/Document.json [hunk 4]
```diff
@@ -7327,6 +7327,214 @@
...
```
Comment (vinyldarkscratch) [inline api/Document.json]:
```suggestion
              "version_added": "12"
```

File: api/Document.json [hunk 5]
```diff
@@ -7327,6 +7327,214 @@
           }
         }
       },
+      "touchcancel_event": {
+        "__compat": {
+          "description": "<code>webglcontextrestored</code> event",
+          "mdn_url": "https://developer.mozilla.org/en-US/docs/Web/API/Document/touchcancel_event",
```
Comment (vinyldarkscratch) [inline api/Document.json]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Document/touchcancel_event",
```

File: api/Document.json [hunk 6]
```diff
@@ -7327,6 +7327,214 @@
...
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },      
+      "touchend_event": {
+        "__compat": {
+          "description": "<code>webglcontextrestored</code> event",
+          "mdn_url": "https://developer.mozilla.org/en-US/docs/Web/API/Document/touchend_event",
```
Comment (vinyldarkscratch) [inline api/Document.json]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Document/touchend_event",
```

File: api/Document.json [hunk 7]
```diff
@@ -7327,6 +7327,214 @@
...
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },      
+      "touchmove_event": {
+        "__compat": {
+          "description": "<code>webglcontextrestored</code> event",
+          "mdn_url": "https://developer.mozilla.org/en-US/docs/Web/API/Document/touchmove_event",
```
Comment (vinyldarkscratch) [inline api/Document.json]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Document/touchmove_event",
```

File: api/Document.json [hunk 8]
```diff
@@ -7327,6 +7327,214 @@
...
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },      
+      "touchstart_event": {
+        "__compat": {
+          "description": "<code>webglcontextrestored</code> event",
+          "mdn_url": "https://developer.mozilla.org/en-US/docs/Web/API/Document/touchstart_event",
```
Comment (vinyldarkscratch) [inline api/Document.json]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Document/touchstart_event",
```

File: api/Element.json [hunk 9]
```diff
@@ -5110,6 +5110,214 @@
           }
         }
       }
+      "touchcancel_event": {
+        "__compat": {
+          "description": "<code>webglcontextrestored</code> event",
+          "mdn_url": "https://developer.mozilla.org/en-US/docs/Web/API/Document/touchcancel_event",
```
Comment (vinyldarkscratch) [inline api/Element.json]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Document/touchcancel_event",
```

File: api/Element.json [hunk 10]
```diff
@@ -5110,6 +5110,214 @@
...
+              "version_added": "22"
+            },
+            "chrome_android": {
+              "version_added": "25"
+            },
+            "edge": {
+              "version_added": "12"
+            },
+            "edge_mobile": {
+              "version_added": "6"
```
Comment (vinyldarkscratch) [inline api/Element.json]:
```suggestion
              "version_added": "12"
```

File: api/Element.json [hunk 11]
```diff
@@ -5110,6 +5110,214 @@
...
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },      
+      "touchend_event": {
+        "__compat": {
+          "description": "<code>webglcontextrestored</code> event",
+          "mdn_url": "https://developer.mozilla.org/en-US/docs/Web/API/Document/touchend_event",
```
Comment (vinyldarkscratch) [inline api/Element.json]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Document/touchend_event",
```

File: api/Element.json [hunk 12]
```diff
@@ -5110,6 +5110,214 @@
...
+              "version_added": "22"
+            },
+            "chrome_android": {
+              "version_added": "25"
+            },
+            "edge": {
+              "version_added": "12"
+            },
+            "edge_mobile": {
+              "version_added": "6"
```
Comment (vinyldarkscratch) [inline api/Element.json]:
```suggestion
              "version_added": "12"
```

File: api/Element.json [hunk 13]
```diff
@@ -5110,6 +5110,214 @@
...
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },      
+      "touchmove_event": {
+        "__compat": {
+          "description": "<code>webglcontextrestored</code> event",
+          "mdn_url": "https://developer.mozilla.org/en-US/docs/Web/API/Document/touchmove_event",
```
Comment (vinyldarkscratch) [inline api/Element.json]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Document/touchmove_event",
```

File: api/Element.json [hunk 14]
```diff
@@ -5110,6 +5110,214 @@
...
+              "version_added": "22"
+            },
+            "chrome_android": {
+              "version_added": "25"
+            },
+            "edge": {
+              "version_added": "12"
+            },
+            "edge_mobile": {
+              "version_added": "6"
```
Comment (vinyldarkscratch) [inline api/Element.json]:
```suggestion
              "version_added": "12"
```

File: api/Element.json [hunk 15]
```diff
@@ -5110,6 +5110,214 @@
...
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },      
+      "touchstart_event": {
+        "__compat": {
+          "description": "<code>webglcontextrestored</code> event",
+          "mdn_url": "https://developer.mozilla.org/en-US/docs/Web/API/Document/touchstart_event",
```
Comment (vinyldarkscratch) [inline api/Element.json]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Document/touchstart_event",
```

File: api/Element.json [hunk 16]
```diff
@@ -5110,6 +5110,214 @@
...
+              "version_added": "22"
+            },
+            "chrome_android": {
+              "version_added": "25"
+            },
+            "edge": {
+              "version_added": "12"
+            },
+            "edge_mobile": {
+              "version_added": "6"
```
Comment (vinyldarkscratch) [inline api/Element.json]:
```suggestion
              "version_added": "12"
```

File: api/Element.json [hunk 17]
```diff
@@ -5110,6 +5110,214 @@
           }
         }
       }
```
Comment (jpmedley) [inline api/Element.json]: This line needs a comma.

Comment (irenesmith) [reply api/Element.json]: Added. Thanks!

**Comment by irenesmith** [comment]
> 
> 
> Thanks for this, @irenesmith ! I fixed the merge conflicts and the lint errors.
> 
> The "description" is wrong for these events. Also, could you point to where the compat data itself comes from, and why you believe it's accurate (at least as accurate as it's reasonable to ascertain)?

I have updated the descriptions.

The BCD data came from the information that was on the event pages before I refactored them but I also confirmed the information with the specifications.

**Pull request merged**